### ISSUE TITLE
update BEDES dictionary to v2.5

### ISSUE THREAD

**Issue by TravisWalterLBNL** [LBNL-ETA/BEDES-Manager#215]
The BEDES atomic terms and list options are currently version 2.4 of the BEDES dictionary, and we need to update to version 2.5. There have been both additions and deletions of atomic terms and list options. Some application mappings are compatible as is, and some need changes.

First, we will need to delete all atomic terms and list options, and also all composite terms, application terms, and application mappings.

We can recreate the atomic terms and list options by reimporting from the exports from the Drupal site.

To recreate the composite terms, application terms, and application mappings, we can just reimport the (currently) 8 application mappings from their exported .csv files. For the 4 application mappings owned by me, I will provide the .csv files (since they required some changes to be compatible with v2.5). For the 4 application mappings owned by the non-LBNL user (Michelle), we can use the admin account to export the mappings to .csv and just reimport those .csv files as is (since they didn't require any changes to be compatible with v2.5). This might require fixing issue #213 first.

When exporting and re-importing application mapping .csv files, the application options aren't preserved. For each of the (currently) 8 applications, the application name will be in the filename of the .csv, the description should be left blank, and the sharing should be set to public.

Also, remember to test issue #200 while working on this issue.

**Comment by wizonesolutions** [comment]
@TravisWalterLBNL Alright, the terms are imported to staging. Can you test importing NYSERDA's applications and make sure you'll be able to restore them as expected? Remember to edit the exports and remove the Composite Term UUID column so that new composite terms get created (it'll fail otherwise).

I took a backup that I can roll back if necessary, and will do the same for production when we update the dictionary there.

**Comment by TravisWalterLBNL** [comment]
I tried to import one my mappings and I'm getting an error message saying `Unrecognized BEDES Term "Premises Level " ...`. At first I thought the issue might be that the error message has an extra space after "Level" (i.e., the correct BEDES term is "Premises Level"), but when I went to search for that BEDES term under "Search for Terms", I couldn't find that term, or any other BEDES terms. So, it looks like the BEDES atomic terms weren't imported correctly somehow. Also, maybe the error message is adding an space to the end of the term name.

**Comment by wizonesolutions** [comment]
Ah, that's strange. Something seems to have happened with the import. Taking a look.

**Comment by wizonesolutions** [comment]
@TravisWalterLBNL Resolved. I import into staging from my local environment, and my env file was still pointing to the old staging database. Have another look if you would.

**Comment by TravisWalterLBNL** [comment]
OK, the atomic terms seem fine now.

I am having one issue with importing app terms with unmapped constrained list options. Say I have an app term "Color" that is a constrained list and it has list options "Red", "Blue", "Green". Can I import this app term and have its list options recorded, even though they aren't mapped? I tried specifying "Red\nBlue\nGreen" in the "BEDES Constrained List Mapping" column in my import .csv, but that gave me a "Each Bedes constrained list mapping needs to be separated by a single '='." error message. I also tried specifying "Red = \nBlue = \nGreen = ", but that gave me "Unrecognized list option "" of constrained list ...". There should be some way to specify unmapped list options for a constrained list app term.

Also, could you please roll back (to a state that is post-"atomic terms are loaded", but pre-"applications are imported and composite terms are created")? I made an application public before importing from .csv and now all the composite terms are showing up as owned by "bedes manager" (instead of "BEDES"), and have "Private" sharing (instead of "Approved"). If I make an app public after importing, everything is fine (i.e., all the composite terms inherit the "BEDES" and "Approved" from the app).

**Comment by wizonesolutions** [comment]
@TravisWalterLBNL I went ahead and did that. I'll take a look at list option importing and what format it is expecting, assuming that it can map blank

 Could you also post or email me your CSV? I'll test that locally as well if I'm still in doubt later.

**Comment by TravisWalterLBNL** [comment]
I attached two versions of the file I was trying to import that has unmapped list options. One specifies the list option mappings as "Red\nBlue\nGreen" and the other specifies "Red = \nBlue = \nGreen = ". For example, see the app term "HistoricalLandmark".
[bsync---no-equals-sign.csv](https://github.com/LBNL-ETA/BEDES-Manager/files/8199812/bsync---no-equals-sign.csv)
[bsync---with-equals-sign.csv](https://github.com/LBNL-ETA/BEDES-Manager/files/8199847/bsync---with-equals-sign.csv)

**Comment by TravisWalterLBNL** [comment]
I opened a new issue (see #217) for importing and exporting unmapped constrained list options, but I think we'll need to fix that issue before we can fully test this one.

**Comment by TravisWalterLBNL** [comment]
Fixed on production.
